File: core/src/main/java/co/fastestpath/api/schedule/route/RouteId.java
package co.fastestpath.api.schedule.route;

public class RouteId {

  private final String value;

  public RouteId(String value) {
    this.value = value;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;

    RouteId stopId = (RouteId) o;

    return value != null ? value.equals(stopId.value) : stopId.value == null;
  }

  @Override
  public int hashCode() {
    return value != null ? value.hashCode() : 0;
  }
}

File: core/src/main/kotlin/co/fastestpath/api/schedule/RouteId.kt
package co.fastestpath.api.schedule

data class RouteId private constructor(private val value: String) {
  companion object {
    private val cache: MutableMap<String, RouteId> = HashMap()

    fun of(value: String): RouteId {
      var routeId = cache[value]
      if (routeId != null) {
        return routeId
      }
      routeId = RouteId(value)
      cache[value] = routeId
      return routeId
    }
  }
}

